src/lib/molecules/ValidatableElements/Elements/ValidatableNumberInput.js

```diff
@@ -86,6 +86,12 @@ const NumberInput = forwardRef((props, ref) => {
     // we use this because need to show field validation after onBlur
     const handleBlur = useCallback(
         (e) => {
+            const { value, validity } = e.target;
+
+            if (validity.badInput && !value.length) {
```

Comment:
The bellow suggestion will cover also `x.` values case
 
```suggestion
            if ((validity.badInput && !value.length) || value.endsWith('.')) {
```

Reply:
A value "x." in the context of a number input typically represents the number x as a floating-point number. The trailing dot suggests that it's a decimal value, even though there are no digits after the decimal point.

In JavaScript programming language, the dot (.) is used to indicate a decimal point. So, "1." is interpreted as the number 1.0 or simply 1.

Let's keep your change like exists.